### ISSUE TITLE
KinematicBody slides on collision box, but doesn't on a plane

### ISSUE THREAD

**Issue by Zylann** [godotengine/godot#26550]
Godot 3.1 beta10

After being confused about what #17893 became, I decided to recreate an issue about the initial point:
If you move a `KinematicBody` with `move_and_slide()`, and make it move towards a box, it will slide on it.
However, if you move towards a plane, it will get stuck in it and doesn't slide. This is unexpected, it should slide just like with the box.

This project demonstrates the problem:
[KinematicBodyOnPlane.zip](https://github.com/godotengine/godot/files/2923711/KinematicBodyOnPlane.zip)

**Comment by abhinavsagar** [comment]
I dont think that is a bug as plane shape is infinite in Godot. To see its effect use some navmesh like covering structured over it .

**Comment by Zylann** [comment]
> I dont think that is a bug as plane shape is infinite in Godot

How is that related to the fact a rigidbody slides but a kinematic doesn't?

**Comment by pouleyKetchoupp** [comment]
Can be still reproduced in 3.2.4 beta 5, but occurs only when using Bullet Physics (default settings). Switching to Godot Physics 3D fixes the issue.

**Comment by akien-mga** [comment]
This seems still reproducible in 3.5 RC 1.
